### PR TITLE
feat: Updated yarn version from 1.x to 3.x.

### PR DESCRIPTION
**Pull request by xdaniortega**
Follows steps defined in issue #40

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Updated gRPC interfaces to 0.14.0
 - Updated emulator SDK to 0.16.2
 - Updated `server-manager` to 0.8.2
+- Updated yarn version to 3.6.3 (berry)
```
Comment (guidanoli) [inline CHANGELOG.md:55]: This should be added in the `Unreleased` section.

File: README.md [hunk 2]
```diff
@@ -164,13 +164,13 @@ As explained above, in order to teleport an asset from the base layer to the exe
...
+| ERC-721           | <ul><li>`address token`,</li><li>`address sender`,</li><li>`uint256 tokenId`,</li><li>standard ABI-encoded fields...</li></ul>                          | <ul><li>`bytes baseLayerData`,</li><li>`bytes execLayerData`</li></ul>                                                           |
+| ERC-1155 (single) | <ul><li>`address token`,</li><li>`address sender`,</li><li>`uint256 tokenId`,</li><li>`uint256 value`,</li><li>standard ABI-encoded fields...</li></ul> | <ul><li>`bytes baseLayerData`,</li><li>`bytes execLayerData`</li></ul>                                                           |
+| ERC-1155 (batch)  | <ul><li>`address token`,</li><li>`address sender`,</li><li>standard ABI-encoded fields...</li></ul>                                                     | <ul><li>`uint256[] tokenIds`,</li><li>`uint256[] values`,</li><li>`bytes baseLayerData`,</li><li>`bytes execLayerData`</li></ul> |
```
Comment (guidanoli) [inline README.md:173]: These changes are unnecessary. The Markdown table is rendered in the exact same way, and is just easier to maintain if you don't care about aligning stuff in the source code.

File: README.md [hunk 3]
```diff
@@ -186,14 +186,14 @@ A voucher can only be executed once the DApp's consensus submits a claim contain
...
+| Ether    | DApp contract  | `withdrawEther(address,uint256)` [:page_facing_up:](./onchain/rollups/contracts/dapp/CartesiDApp.sol)                                            |
+| ERC-20   | Token contract | `transfer(address,uint256)` [:page_facing_up:](https://eips.ethereum.org/EIPS/eip-20#methods)                                                    |
+| ERC-20   | Token contract | `transferFrom(address,address,uint256)` [:page_facing_up:](https://eips.ethereum.org/EIPS/eip-20#methods) [^1]                                   |
+| ERC-721  | Token contract | `safeTransferFrom(address,address,uint256)` [:page_facing_up:](https://eips.ethereum.org/EIPS/eip-721#specification)                             |
+| ERC-721  | Token contract | `safeTransferFrom(address,address,uint256,bytes)` [:page_facing_up:](https://eips.ethereum.org/EIPS/eip-721#specification) [^2]                  |
+| ERC-1155 | Token contract | `safeTransferFrom(address,address,uint256,uint256,data)` [:page_facing_up:](https://eips.ethereum.org/EIPS/eip-1155#specification)               |
```
Comment (guidanoli) [inline README.md:196]: Same here.

File: onchain/.gitignore [hunk 4]
```diff
@@ -4,3 +4,12 @@ node_modules
 .DS_Store
 cache
 out
+
+# Follows https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored and not using Zero-Installs feature
```
Comment (guidanoli) [inline onchain/.gitignore:8]: Nice one!

**Comment by guidanoli** [comment]
Oops, sorry!

**Comment by guidanoli** [comment]
I also wanted to comment about the commit messages.

> feat: Updated yarn version from 1.x to 3.x.
> docs: Updated yarn installation link in README.

Please prefer present-tense, imperative-style commit messages, like below. Also, we usually don't capitalize the first word in the commit message. But that's totally up to you.

> feat: Update yarn version from 1.x to 3.x.
> docs: Update yarn installation link in README.

**Comment by guidanoli** [comment]
Also, you don't need to add labels or milestones to PRs, or add them to the Rollups Unit project. PRs are tracked by the issues they are linked to. :-)
You just need to set reviewer(s), assignee(s), and link the PR to the corresponding issue.

File: onchain/.gitignore (hunk 4 above)
Comment (xdaniortega) [reply onchain/.gitignore:8]: Thanks!

**Comment by xdaniortega** [comment]
Rebased on top of main

**Comment by xdaniortega** [comment]
After rebasing on top of main, `CartesiDapp.t.sol` was not updated from main into this branch. Weird that there was no warning also from Github for merging into main, only CI noticed.
Manually copied from main to this branch, now all tests pass.

We'll leave plug&play from yarn for another issue, so we can introduce yarn v3 with no problems and fully compatible with existing dependencies in `node_modules` folder.

**Comment by guidanoli** [comment]
> After rebasing on top of main, `CartesiDapp.t.sol` was not updated from main into this branch. Weird that there was no warning also from Github for merging into main, only CI noticed. Manually copied from main to this branch, now all tests pass.

Here's the routine I use to rebase branches.

```
git fetch --prune
git checkout <feature-branch>
git rebase -i origin/<base-branch>
```

**Comment by guidanoli** [comment]
> We'll leave plug&play from yarn for another issue, so we can introduce yarn v3 with no problems and fully compatible with existing dependencies in node_modules folder.

I'm afraid this is a blocker. We should first make sure everything works smoothly before committing to this migration.

**Comment by xdaniortega** [comment]
> > After rebasing on top of main, `CartesiDapp.t.sol` was not updated from main into this branch. Weird that there was no warning also from Github for merging into main, only CI noticed. Manually copied from main to this branch, now all tests pass.
> 
> Here's the routine I use to rebase branches.
> 
> ```
> git fetch --prune
> git checkout <feature-branch>
> git rebase -i origin/<base-branch>
> ```

I did this exact procedure but it was not detecting the file, idk why. Is it ok anyway, right?



> > We'll leave plug&play from yarn for another issue, so we can introduce yarn v3 with no problems and fully compatible with existing dependencies in node_modules folder.
> 
> I'm afraid this is a blocker. We should first make sure everything works smoothly, otherwise we'll be committing into a half-baked solution.

True! I just left this PR to update the yarn version and everything works perfectly. I left my remaining progress with PnP at #92 .

**Comment by guidanoli** [comment]
> I did this exact procedure but it was not detecting the file, idk why. Is it ok anyway, right?

If you're able to reproduce this again, we can schedule a meeting so we can investigate this further.

**Comment by guidanoli** [comment]
> True! I just left this PR to update the yarn version and everything works perfectly. I left my remaining progress with PnP at https://github.com/cartesi/rollups-contracts/issues/92 .

Nice. Hey, maybe we can search for web3 projects that use Yarn modern and see how they're resolving Solidity dependencies. :-)

**Comment by guidanoli** [comment]
Seems like the `prepack` script is still supported by yarn modern.
https://yarnpkg.com/advanced/lifecycle-scripts#prepack-and-postpack

**Comment by guidanoli** [comment]
Hey, @xdaniortega.
I've run the commands from the step-by-step guide and ended up with different files.
Please make sure you are running all of them in order, and committing all the files created.

**Comment by xdaniortega** [comment]
> Hey, @xdaniortega.
> I've run the commands from the step-by-step guide and ended up with different files.
> Please make sure you are running all of them in order, and committing all the files created.

Hi! Interesting, what files are you getting different? Maybe just ` onchain/rollups/lib/openzeppelin-contracts (new) |     1 +`? Be sure you're setting `nodeLinker: node-modules` at `yarn.yaml`.

**Comment by xdaniortega** [comment]
Rebased on top of main

File: onchain/.yarnrc.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+nodeLinker: node-modules
+
+yarnPath: .yarn/releases/yarn-3.6.4.cjs
```
Comment (xdaniortega) [inline onchain/.yarnrc.yml:3]: @guidanoli Do you wave the same `uarnPath: .yarn/releases/yarn-3.6.4.cjs` generated line?

Comment (guidanoli) [reply onchain/.yarnrc.yml:3]: Yes.

File: CHANGELOG.md
Comment (guidanoli) [inline CHANGELOG.md:1]: As this yarn update will not affect the end user, we don't need to add this to the changelog. :-)

**Comment by xdaniortega** [comment]
This PR is ready for final review.

**Review by guidanoli — APPROVED** [review]

**Pull request merged**